var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -736,15 +789,24 @@ def install(self, spec, prefix):
         if jobs > 64 and spec.satisfies("@:1.58"):
             jobs = 64
 
-        # Windows just wants a b2 call with no args
-        b2_options = []
-        if not self.spec.satisfies("platform=windows"):
-            path_to_config = "--user-config=%s" % os.path.join(
-                self.stage.source_path, "user-config.jam"
-            )
-            b2_options = ["-j", "%s" % jobs]
-            b2_options.append(path_to_config)
+        if self.spec.satisfies("platform=windows"):
 
+            def is_64bit():
+                # TODO: This method should be abstracted to a more general location
+                #  as it is repeated in many places (msmpi.py for one)
+                return "64" in str(self.spec.target.family)
+
+            b2_options = [f"--prefix={self.prefix}", f"address-model={64 if is_64bit() else 32}"]
+            if not self.spec.satisfies("+python"):
+                b2_options.append("--without-python")
```

Comment:
You cannot combine `--with-<opt>` and `--without-<opt>`, so since the rest of the package file uses `--without-<opt>`, then this must be `--without-python` if the spec does not satisfy `+python`

Reply:
Yes, appears to be a Windows difference which is that these options are initially provided during bootstrapping on linux, but on windows, the bootstrapping process does not accept the `--with` or `--without` flags, they are only provided to `b2`, which, when I tried to mix `with` and `without` flags, I got an error message on Windows.

Reply:
So if I understand:

```
./b2 --without-python --without-icu #works
./b2 --with-python --without-icu # fails
./b2 --with-python # works, and doesn't use icu
```

Would you be able to include the error message here?

Reply:
The wording of that error message makes it sound like it got something like `--with-python --without-python` (i.e. `--with-x` and `--without-x` where X is the same in both cases) - are you not seeing anything like that in the build log?

Reply:
I agree the message wording is ambiguous, however, I get the same error message with a much simpler `b2` command where I mix one `--with` and one `--without` flag (split for readability): 

```
.\b2.exe 
--clean
--prefix=C:\spack\opt\spack\windows-windows10.0.19045-skylake\msvc-19.39.33522\boost-1.82.0-t4ec3auwcpgtjnulmbbwwydlxtacntzs
address-model=64
--user-config=C:\spack\.staging\boost-1.82.0-t4ec3au\spack-src\user-config.jam
variant=release
link=static,shared
--layout=system
runtime-link=shared
--with-atomic
--without-chrono
cxxstd=11
visibility=hidden
```

Reply:
Thanks for adding that!